src/uploader.js

```diff
@@ -63,23 +63,27 @@ const useWidget = (
   const attributes = useDeepMemo(() => propsToAttr(options), [options])
 
   useDeepEffect(() => {
-    if (render.current) {
-      render.current = false
-
-      return () => {
-        delete window.UPLOADCARE_LOCALE
-        delete window.UPLOADCARE_LOCALE_TRANSLATIONS
-        delete window.UPLOADCARE_LOCALE_PLURALIZE
-      }
+    if (!firstRender.current) {
+      window.UPLOADCARE_LOCALE = locale
+      window.UPLOADCARE_LOCALE_TRANSLATIONS = localeTranslations
+      window.UPLOADCARE_LOCALE_PLURALIZE = localePluralize
+
+      uploadcare.plugin((internal) =>
+        internal.locale.rebuild({
+          locale,
+          localeTranslations,
+          localePluralize
+        })
+      )
     }
 
-    uploadcare.plugin((internal) =>
-      internal.locale.rebuild({
-        locale,
-        localeTranslations,
-        localePluralize
-      })
-    )
+    firstRender.current = false
+
+    return () => {
+      delete window.UPLOADCARE_LOCALE
+      delete window.UPLOADCARE_LOCALE_TRANSLATIONS
+      delete window.UPLOADCARE_LOCALE_PLURALIZE
+    }
```

Comment:
Since `window.UPLOADCARE_` variables aren't set on the first render, you take for granted that users will manually set them it in their code **and** also pass them in here. While that's acceptable, this would be a good opportunity to allow users to define them only via react-widget's props and set the global variables on the first render.

Reply:
Another thing to note is that passing _only a single one_ of these props gets rid of all the other locale settings the user defined via their global config.

It would be safer to check if each prop (locale, localeTranslations, localePluralize) is defined before reassigning / deleting the corresponding global variable.

Reply:
> It would be safer to check if each prop (locale, localeTranslations, localePluralize) is defined before reassigning / deleting the corresponding global variable.

yes, I totally agree with that, nice catch 🙌🏻

> Since window.UPLOADCARE_ variables aren't set on the first render.

window.UPLOADCARE_ variables are set on first render at 
https://github.com/uploadcare/react-widget/blob/master/src/index.js#L21-L28
before importing `uploadcare-widget`, but I can move that logic in one place, will test